site/en/docs/extensions/mv3/tut_analytics/index.md

```diff
@@ -1,166 +1,261 @@
-**\_\_utm.gif** if everything is set up correctly.
 
-{% Img src="image/BrQidfK9jaQyIHwdw91aVpkPiib2/lPW0DjyEMWYnhPc336QN.png",
-       alt="Developer Tools window showing the __utm.gif request", height="418", width="683" %}
+```js
+const SESSION_EXPIRATION_IN_MIN = 30;
+
+async function getOrCreateSessionId() {
+  // Store session in memory storage
+  let {sessionData} = await chrome.storage.session.get("sessionData");
+  // Check if session exists and is still valid
+  const currentTimeInMs = Date.now();
+  if (sessionData && sessionData.timestamp) {
+    // Calculate how long ago the session was last updated
+    const durationInMin = (currentTimeInMs - sessionData.timestamp) / 60000;
+    // Check if last update lays past the session expiration threshold
+    if (durationInMin > SESSION_EXPIRATION_IN_MIN) {
+      // Delete old session id to start a new session
+      sessionData = null;
+    } else {
+      // Update timestamp to keep session alive
+      sessionData.timestamp = currentTimeInMs;
+      await chrome.storage.session.set({ sessionData });
+    }
+  }
+  if (!sessionData) {
+    // Create and store a new session
+    sessionData = {
+      session_id: currentTimeInMs.toString(),
+      timestamp: currentTimeInMs.toString(),
+    };
+    await chrome.storage.session.set({ sessionData });
+  }
+  return sessionData.session_id;
+}
+```
 
-## Tracking events {: #toc-tracking-events }
 
-By configuring event tracking, you can determine which parts of your extension your users interact
-with the most. For example, if you have three buttons users may click:
+The following example adds `session_id` and `engagement_time_msec` to the previous button click event request. For `engagement_time_msec` you can provide a default value of `100 ms`. 
 
-```html
-<button id='button1'>Button 1</button>
-<button id='button2'>Button 2</button>
-<button id='button3'>Button 3</button>
+```js
+const GA_ENDPOINT = "https://www.google-analytics.com/mp/collect";
+const MEASUREMENT_ID = `G-...`;
+const API_SECRET = `...`;
+const DEFAULT_ENGAGEMENT_TIME_IN_MSEC = 100;
+
+fetch(
+`${GA_ENDPOINT}?measurement_id=${MEASUREMENT_ID}&api_secret=${API_SECRET}`,
+  {
+    method: "POST",
+    body: JSON.stringify({
+      client_id: getOrCreateClientId(),
+      events: [
+        {
+          name: "button_clicked",
+          params: {
+            session_id: await this.getOrCreateSessionId(),
+            engagement_time_msec" DEFAULT_ENGAGEMENT_TIME_IN_MSEC,
+            id: "my-button",
+          },
+        },
+      ],
+    }),
+  }
+);
 ```
 
-Write a function that sends click events to Google Analytics:
+The event will be displayed as follows in the Google Analytics real-time report.
+
+{% Img src="image/6hHqS5auVgWhN0cQNQztaJx5w4M2/AQqSJb9VqalWfrQ6xSXM.png", alt="Realtime events in Google Analytics. ", width="396", height="414" %}
+
+
+## Tracking page views in tabs, popup and sidepanel  {: #toc-tracking-pageviews }
```

Comment:
```suggestion
## Tracking page views in tabs, popup, and side panel  {: #toc-tracking-pageviews }

```
Which tabs are these? Extension HTML pages? Tabs we've injected a contents scripts into?

Reply:
"Extension HTML pages" - what's the correct way to refer to these?